src/AudioUnitSDK/AUEffectBase.cpp

```diff
@@ -251,12 +251,12 @@ OSStatus AUEffectBase::ProcessScheduledSlice(void* inUserData, UInt32 /*inStartF
 
 	UInt32 channelSize = inSliceFramesToProcess * mBytesPerFrame;
 	// fix the size of the buffer we're operating on before we render this slice of time
-	for (UInt32 i = 0; i < inputBufferList.mNumberBuffers; i++) {
+	for (UInt32 i = 0, nBuf = inputBufferList.mNumberBuffers; i < nBuf; i++) {
```

Comment:
what is good about this?

Reply:
inputBufferList.mNumberBuffers may be reloaded from memory. But I can change it back if it doesn't matter

Reply:
Okay understood and yes true. We shall evaluate and consider, but not sure the extra complexity to address this everywhere it could occur is warranted since we have never seen it show up as a concern in performance profiles, and the number of similar reads and writes with the same input objects of their audio sample data in most cases dwarfs reads of this value, and do not have a means to avoid it.